addon/components/ember-collection.js

```diff
@@ -27,59 +25,56 @@ function formatStyle(pos, width, height) {
 export default Ember.Component.extend({
   layout: layout,
 
-  // Utility to get attribute value which may or may not be wrapped in mut helper.
-  // returns defaultValue if attribute not defined or defined as null or undefined
-  _maybeMutAttr(key, defaultValue) {
-    if (this.attrs == null) { return defaultValue; }
-    var obj = this.attrs[key];
-    if (obj == null) { return defaultValue; }
-    obj = getMutValue(obj);
-    obj = (obj == null) ? defaultValue : obj;
-    return obj;
-  },
   init() {
-    this._super();
+    // State pulled from attrs is prefixed with an underscore
+    // so that there's no chance of shadowing the attrs proxy.
+    this._buffer = undefined;
+    this._cellLayout = undefined;
+    this._items = undefined;
+    this._offsetX = undefined;
+    this._offsetY = undefined;
+    this._width = undefined;
+    this._height = undefined;
+    
     // this.firstCell = undefined;
     // this.lastCell = undefined;
     // this.cellCount = undefined;
-    this.offsetX = 0;
-    this.offsetY = 0;
-    this.width = 0;
-    this.height = 0;
     this.contentElement = undefined;
-    Ember.set(this, 'cells', Ember.A([]));
-    this.cellMap = Object.create(null);
+    this._cells = [];
+    this._cellMap = Object.create(null);
+
+    this._super();
```

Comment:
it would be useful to figure out exactly which properties break it, and order accordingly. 

Leaving inline docs would also be great.